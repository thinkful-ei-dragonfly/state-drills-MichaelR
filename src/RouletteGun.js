import React from 'react';
import './RouletteGun.css';

class RouletteGun extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      chamber: null,
      spinningTheChamber: false,
    }
  }

  handleClick = () => {
    this.setState({
      spinningTheChamber: true
    })
    this.timeout = setTimeout(() => {
      let newChamber = Math.ceil(Math.random() * 8)
      this.setState({
        chamber: newChamber,
        spinningTheChamber: false
      })
    }, 2000);
  }

  componentWillUnmount() {
    clearTimeout(this.timeout);
  }

  render() {
    let message = `Let's play Russian Roulette`;
    if (this.state.spinningTheChamber) {
      message = 'spinning the chamber and pulling the trigger! ...'
    } else if (this.state.chamber === this.props.bulletInChamber) {
      message = 'BANG!!!!';
    } else if (this.state.chamber) {
      message = `You're safe!`;
    }

    return (
      <div className="RouletteGun">
        <p>{message}</p>
        <button onClick={this.handleClick}>Pull the Trigger!</button>
      </div>
    )
  };
}

RouletteGun.defaultProps = ({
  bulletInChamber: 8
})

export default RouletteGun;