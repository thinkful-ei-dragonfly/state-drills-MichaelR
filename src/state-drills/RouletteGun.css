.RouletteGun {
  margin: 20px;
}