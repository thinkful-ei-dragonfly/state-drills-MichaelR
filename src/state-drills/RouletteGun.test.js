import React from 'react';
import renderer from 'react-test-renderer';

import RouletteGun from './RouletteGun';

describe('<RouletteGuns />', () => {
  it('renders without crashing [snapshot test]', () => {
    // Render this component as JSON
    const tree = renderer.create(<RouletteGun />).toJSON();
    // Checks wether this matches the previous snapshot
    // The previous snapshot will be stored in __snapshots__/RouletteGuns .test.js
    expect(tree).toMatchSnapshot();
  })
})