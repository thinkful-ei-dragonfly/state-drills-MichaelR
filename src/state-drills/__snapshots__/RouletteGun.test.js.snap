// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouletteGuns /> renders without crashing [snapshot test] 1`] = `
<div
  className="RouletteGun"
>
  <p>
    Let's play Russian Roulette
  </p>
  <button
    onClick={[Function]}
  >
    Pull the Trigger!
  </button>
</div>
`;
